packages/common/src/ide/inMemoryTextDocument/InMemoryTextLine.ts

```diff
@@ -4,11 +4,10 @@ import type { TextLine } from "../../types/TextLine";
 import { getLeadingWhitespace, getTrailingWhitespace } from "../../util/regex";
 
 export class InMemoryTextLine implements TextLine {
+  readonly isEmptyOrWhitespace: boolean;
   readonly range: Range;
   readonly rangeIncludingLineBreak: Range;
-  readonly firstNonWhitespaceCharacterIndex: number;
-  readonly lastNonWhitespaceCharacterIndex: number;
-  readonly isEmptyOrWhitespace: boolean;
+  readonly rangeTrimmed: Range;
```

Comment:
update from meet-up: let's make this `undefined` if `isEmptyOrWhitespace` is `true`